src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/PackageArgumentParserUtility.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.CommandLine.Parsing;
+using NuGet.CommandLine.XPlat.Commands.Package;
+using NuGet.Versioning;
+
+namespace NuGet.CommandLine.XPlat.Utility
+{
+    internal static class PackageArgumentParserUtility
+    {
+        public static IReadOnlyList<PackageArgument<NuGetVersion>> ParseWithVersion(ArgumentResult result)
+        {
+            return PackageArgument<NuGetVersion>.Parse(
+                result,
+                NuGetVersion.TryParse,
+                versionString => Messages.Error_InvalidVersion(versionString),
```

Comment:
```suggestion
                Messages.Error_InvalidVersion,
```

Creating an anonymous method always causes additional memory allocations. Passing an existing method with the right signature will avoid that.